x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.ts

```diff
@@ -5,81 +5,145 @@
-        body: {
-          action: '713085d6-ab45-4e9e-b41d-96563cafdd97',
-        },
-      });
-    }
+    isolationRequestHandler(endpointContext, true)
   );
 
   // perform UN-isolate
   router.post(
     {
-      path: `/api/endpoint/unisolate`,
-      validate: {
-        body: schema.object({
-          agent_ids: schema.nullable(schema.arrayOf(schema.string())),
-          endpoint_ids: schema.nullable(schema.arrayOf(schema.string())),
-          alert_ids: schema.nullable(schema.arrayOf(schema.string())),
-          case_ids: schema.nullable(schema.arrayOf(schema.string())),
-          comment: schema.nullable(schema.string()),
-        }),
-      },
+      path: UNISOLATE_HOST_ROUTE,
+      validate: HostIsolationRequestSchema,
       options: { authRequired: true },
     },
-    async (context, req, res) => {
-      if (
-        (req.body.agent_ids === null || req.body.agent_ids.length === 0) &&
-        (req.body.endpoint_ids === null || req.body.endpoint_ids.length === 0)
-      ) {
-        return res.badRequest({
-          body: {
-            message: 'At least one agent ID or endpoint ID is required',
+    isolationRequestHandler(endpointContext, false)
+  );
+}
+
+export const isolationRequestHandler = function (
+  endpointContext: EndpointAppContext,
+  isolate: boolean
+): RequestHandler<
+  unknown,
+  unknown,
+  TypeOf<typeof HostIsolationRequestSchema.body>,
+  SecuritySolutionRequestHandlerContext
+> {
+  return async (context, req, res) => {
+    if (
+      (!req.body.agent_ids || req.body.agent_ids.length === 0) &&
+      (!req.body.endpoint_ids || req.body.endpoint_ids.length === 0)
+    ) {
+      return res.badRequest({
+        body: {
+          message: 'At least one agent ID or endpoint ID is required',
+        },
+      });
+    }
+
+    // only allow admin users
+    const user = endpointContext.service.security?.authc.getCurrentUser(req);
+    if (!UserCanIsolate(user?.roles)) {
+      return res.forbidden({
+        body: {
+          message: 'You do not have permission to perform this action',
+        },
+      });
+    }
+
+    // isolation requires plat+
+    if (isolate && !endpointContext.service.getLicenseService()?.isPlatinumPlus()) {
+      return res.forbidden({
+        body: {
+          message: 'Your license level does not allow for this action',
+        },
+      });
+    }
+
+    // translate any endpoint_ids into agent_ids
+    let agentIDs = req.body.agent_ids?.slice() || [];
+    if (req.body.endpoint_ids && req.body.endpoint_ids.length > 0) {
+      const newIDs = await getAgentIDsForEndpoints(req.body.endpoint_ids, context, endpointContext);
+      agentIDs = agentIDs.concat(newIDs);
+    }
+    agentIDs = [...new Set(agentIDs)]; // dedupe
+
+    // create an Action ID and dispatch it to ES & Fleet Server
+    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    const actionID = uuid.v4();
+    let result;
+    try {
+      result = await esClient.index({
```

Comment:
`.create` requires an explicit document ID. We don't want to do that

`.index` will auto-create the document for you.

[esClient create reference](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_create) which calls the `/_create/<_id>` route under the hood.

[Elasticsearch document creation reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html)

since we don't want to specify an `_id` explicitly at creation, we want the `POST /<index>/_doc/` route, which is equivalent to an **Index** call